src/training_raster_clipper/implementation/nbertaud.py

```diff
@@ -0,0 +1,194 @@
+    ClassifiedSamples,
+    FeatureClassNameToId,
+    PolygonMask,
+    ResolutionType,
+)
+
+
+def load_feature_polygons(input_path: Path) -> GeoDataFrame:
+    gdf = geopandas.io.file.read_file(input_path)
+    gdf = gdf.to_crs(32631)
+    return gdf
+
+
+def load_sentinel_data(
+    sentinel_product_location: Path,
+    resolution: ResolutionType,
+    band_names: tuple[BandNameType, ...],
+) -> xr.DataArray:
+    """Loads sentinel product
+
+    Example input path: `S2A_MSIL2A_20221116T105321_N0400_R051_T31TCJ_20221116T170958.SAFE`
+
+    Args:
+        sentinel_product_location (Path): Location of the .SAFE folder containing a Sentinel-2 product.
+
+    Returns:
+        xr.DataArray: A DataArray containing the 3 RGB bands from the visible spectrum
+    """
+
+    # Find the path of the folder with the correct resolution
+    spatial_resolution_location = list(sentinel_product_location.glob(f"GRANULE/*/IMG_DATA/R{resolution}m"))[0]
+
+    # Find the path of the bands
+    band_locations = {}
+    for band_name in band_names:
+        band_locations[band_name] = list(spatial_resolution_location.glob(f"*_{band_name}_*"))[0]
+
+    # Read the band raster
+    combined_data_array = []
+    for band_name, raster in band_locations.items():
+        band_data_array = rioxarray.open_rasterio(raster)
+        band_data_array = band_data_array.assign_coords(coords={"band": [band_name]})
+        combined_data_array.append(band_data_array)
+
+    # Combine all the band in the same xarray
+    combined_data_array = xr.concat(combined_data_array, "band")
+
+    # Normalize the xarray
+    SPECIAL_VALUE = 0
+    RADIO_ADD_OFFSET = -1000
+    QUANTIFICATION_VALUE = 10000
+    normalized_data_array = combined_data_array.where(combined_data_array != SPECIAL_VALUE, np.nan)
+
+    normalized_data_array = (normalized_data_array + RADIO_ADD_OFFSET) / QUANTIFICATION_VALUE
+
+    return normalized_data_array
+
+
+def rasterize_geojson(
+    data_array: xr.DataArray,
+    training_classes: GeoDataFrame,
+) -> tuple[PolygonMask, FeatureClassNameToId]:
+    """Burns a set of vectorial polygons to a raster.
+
+    See https://gis.stackexchange.com/questions/316626/rasterio-features-rasterize
+
+    Args:
+        data_array (xr.DataArray): The Sentinel raster, from which data is taken, such as the transform or the shape.
+        training_classes (GeoDataFrame): The input set of classified multipolygons to burn
+
+    Returns:
+        xr.DataArray: A mask raster generated from the polygons, representing the same geographical region as the source dataarray param
+                      0 where no polygon were found, and integers representing classes in order of occurence in the GeoDataFrame
+    """
+
+    # Find transformation
+    transform = np.array(data_array.spatial_ref.GeoTransform.split(" "), dtype=float)
+    transform = rasterio.transform.Affine.from_gdal(*transform)
+
+    # Find array shape
+    array_shape = data_array.shape[1:]
```

Comment:
When reading this code, I know instictively that the shape is of len 3, and constituted of band, y, x, so that when you say `shape[1:]` what you really say is "shape of y and x dimensions, ignore the band"

You can be more explicit by saying for instance `data_array.isel(band=0).shape`

"isel with 0" means "take the first slice, whatever", then use the slice of it

so basically `1` became `band`

One of the main goal of xarray is to name the dimensions